auth/auth.guard.ts

```diff
@@ -1,36 +1,52 @@
-/*
- * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
- * Copyright 2023 Xyna GmbH, Germany
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *  http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
- */
-import { Injectable } from '@angular/core';
-import { ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+// /*
+//  * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+//  * Copyright 2023 Xyna GmbH, Germany
+//  *
+//  * Licensed under the Apache License, Version 2.0 (the "License");
+//  * you may not use this file except in compliance with the License.
+//  * You may obtain a copy of the License at
+//  *
+//  *  http://www.apache.org/licenses/LICENSE-2.0
+//  *
+//  * Unless required by applicable law or agreed to in writing, software
+//  * distributed under the License is distributed on an "AS IS" BASIS,
+//  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  * See the License for the specific language governing permissions and
+//  * limitations under the License.
+//  * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+//  */
+
+
+import { inject, Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivateFn, RouterStateSnapshot } from '@angular/router';
 
 import { Observable } from 'rxjs';
 
 import { AuthService } from './auth.service';
 
 
 @Injectable()
-export class AuthGuard  {
+export class AuthGuardService {
 
     constructor(private readonly authService: AuthService) {
     }
 
-
-    canActivate(activatedRoute: ActivatedRouteSnapshot, routerState: RouterStateSnapshot): Observable<boolean> {
-        return this.authService.queryAuthentication(routerState.url);
+    canActivate(activatedRoute: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
+        return this.authService.queryAuthentication(state.url);
     }
 }
+
+export const AuthGuard: CanActivateFn = (route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> => inject(AuthGuardService).canActivate(route, state);

```

Comment:
I would rather like all these De-/Activate-functions to have a uniform structure, as you did it for the `RedirectGuard`:

```
authGuardCanActivate
redirectGuardCanActivate
redirectGuardCanDeactivate
rightGuardCanActivate
```

Although this would introduce some more breaking changes, we give these functions a more consistent look and we're closer to the Angular convention.